src/network/networkd-link.c

```diff
@@ -199,6 +199,21 @@ static bool link_proxy_arp_enabled(Link *link) {
         return true;
 }
 
+static bool link_ipv6_proxy_ndp_enabled(Link *link) {
```

Comment:
This got obsoleted, as the value for the `proxy_ndp` flag is now deduced from `network->n_ipv6_proxy_ndp_addresses`.
The logic for this check was moved into the `ipv6_proxy_ndp_is_needed(Link*)` function.